incubator/hnc/internal/validators/object.go

```diff
@@ -246,7 +246,13 @@ func (o *Object) hasConflict(inst *unstructured.Unstructured) (bool, []string) {
 	// Get a list of conflicting descendants if there's any.
 	for _, desc := range descs {
 		if o.Forest.Get(desc).HasSourceObject(gvk, nm) {
-			conflicts = append(conflicts, desc)
+			// If the user have chosen not to propagate the object to this descendant,
+			// there shouldn't be any conflict reported here
+			nsLabels := o.Forest.Get(inst.GetNamespace()).GetLabels()
+			isMatch, _ := selectors.DoSelectorsMatch(inst, nsLabels)
```

Comment:
Having another look, I think `DoSelectorsMatch` is actually a bit confusing. Maybe just `selectors.ShouldPropagate` would be easier to read?

Also you can do this as a one line:

```
if conflict, _ := selectors.ShouldPropagate(inst, nsLabels); conflict {
  conflicts = append(conflicts, desc)
}
```

Reply:
We already have some other functions called 'shouldPropagate()', do you think this will make the function names confusing?

Reply:
IMO it's fine to have similarly named functions in different packages. `shouldPropagate` in a reconciler might be making the overall decision, while `selectors.ShouldPropagate` is only about whether the _selectors_ think it's ok to propagate.

Reply:
Makes sense, I'm changing it to ShouldPropagate